src/libostree/ostree-kernel-args.c

```diff
@@ -163,6 +163,49 @@ strcmp0_equal (gconstpointer v1, gconstpointer v2)
   return g_strcmp0 (v1, v2) == 0;
 }
 
+/* Split string with whitespace, but keep quotes.
+   For example, "test=\"1 2\"" will be saved as whole.
+ */
+static char **
+split_whitespace (const char *str)
+{
+  gboolean quoted = FALSE;
+  g_return_val_if_fail (str != NULL, NULL);
+
+  GPtrArray *strv = g_ptr_array_new ();
+
+  guint len = strlen (str);
+  guint start;
+  // Skip first whitespaces
+  for (start = 0; start < len && str[start] == ' '; start++)
+    {
+    }
+  for (guint i = start; str[i] != '\0'; i++)
+    {
+      if (str[i] == '"')
+        quoted = !quoted;
+      else if (str[i] == ' ' && !quoted)
+        {
+          g_ptr_array_add (strv, g_strndup (str + start, i - start));
+          start = i + 1;
+        }
+
+      if (i == (len - 1))
+        {
+          if (!quoted)
+            g_ptr_array_add (strv, g_strndup (str + start, len - start));
+          else
+            {
+              g_printerr ("Error: Missing terminating quote in '%s'.\n", str);
```

Comment:
We shouldn't print to stderr from a library as a general rule; it's either a fatal error (`GError`) or it's something we accept and log with `g_debug()` by default.

The question here is what does the Linux kernel do?  If it accepts it then we should too.  I suspect it just accepts it.